### DISCUSSION TITLE
How to set maximum lines of rendered HTML page?

### DISCUSSION THREAD

**Discussion by tedkong** [eclipse-birt/birt#1513] [Q&A] [answered]
I tested with BIRT 4.7.0, and found that I could render a specific page if I want using the following method in case the report size is huge.

renderTask.setPageNumber( 7);

It displayed 10,000 lines for each page as default.
Can I change the number of lines displayed in one page using some method?

**Comment by speckyspooky** [comment]
You can try to define a very large page on the master page and define the page hight very hugh.
SO you will be able to display more like 10.000 lines per page.

The minimum is limited by the maximum page height and dependend on your font size and your table structure with your margins and so on.
The maximum height is in the near of 34.000 cm = 340m.

**Reply by tedkong** [reply]
Hi @speckyspooky 
Thanks a lot for your advice.
May I know the sample codes how I can set the value?

**Reply by speckyspooky** [reply, marked as answer]
You can set the page height of the master page directly into your report.
(No special coding needed.)

### Setup of the master page height
![mp-height](https://github.com/eclipse-birt/birt/assets/41593722/650c31dd-ffc7-49cc-a6b6-3dda3a004238)

**Comment by hvbtup** [comment]
I'm not using HTML output, but maybe you should look at the layout preference for the report.

![grafik](https://github.com/eclipse-birt/birt/assets/7644776/4d60d2c8-c70c-48c8-92ec-2d48a78614a0)

You could try using "Fixed Layout" and a realistic page height for the master page.

I guess BIRT will create smaller HTML pages then.

**Comment by tedkong** [comment]
Thank you @speckyspooky  and @hvbtup for your comments.

Finally, I found one solution to change the value using a property value 'pageBreakInterval' inside the report design file.

Thanks again for your kind answers.
